model/flow/cluster.go

```diff
@@ -45,6 +45,8 @@ func (clusters ClusterList) Assignments() AssignmentList {
 
 // NewClusterList creates a new cluster list based on the given cluster assignment
 // and the provided list of identities.
+// The caller must ensure each assignment contains identities ordered in canonical order, so that
+// each cluster in the returned cluster list is ordered in canonical order as well.
 func NewClusterList(assignments AssignmentList, collectors IdentityList) (ClusterList, error) {
```

Comment:
Can we check for canonical ordering here and return an error if not?

Reply:
Addressed [here](https://github.com/onflow/flow-go/pull/2140/commits/fbd30bf39e6aab8a678319393a9a4140d730815e)